src/main/java/vct/col/rewrite/SimplifyQuantifiedRelations.scala

```diff
@@ -34,6 +34,20 @@ class SimplifyQuantifiedRelations(source: ProgramUnit) extends AbstractRewriter(
     case other => Seq(other)
   }
 
+  def splitSelect(select: ASTNode, main: ASTNode): (Seq[ASTNode], ASTNode) = {
```

Comment:
I think I have a vague idea about what this pass does. Just to be sure, maybe I missed something: what is the reason this cannot be implemented as one or more rewrite rules? It seems to handle some complex cases, but the example on line 194 seems to be handleable as a rewrite rule.

Reply:
This is indeed a little sad, in fact I would have liked this entire pass to be rewrite rules, but it doesn't quite work. This splitSelect indeed would work as rewrite rules, however here it's used to compose with itself and I don't think we want to run jspec rules within a pass. Example:

```
forall int i=0..N; forall int j=0..M; i*M+j < size
==> the inner part is rewritten
forall int i=0..N; 0<M ==> (i*M+(M-1)) < size
==> the select part is made part of the condition
forall int i=0..N && 0<M; (i*M+(M-1)) < size
==> the outer part is rewritten (now possible because the body is no longer an implies)
(0<N && 0<M) ==> ((N-1)*M+(M-1)) < size
```